testssl.sh

```diff
@@ -16556,6 +16588,8 @@ find_openssl_binary() {
      $OPENSSL enc -aes-256-gcm -K 0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef -iv 0123456789abcdef01234567  > /dev/null 2> /dev/null <<< "test"
      [[ $? -eq 0 ]] && HAS_AES256_GCM=true
 
+     [[ "$(echo -e "\x78\x9C\xAB\xCA\xC9\x4C\xE2\x02\x00\x06\x20\x01\xBC" | $OPENSSL zlib -d 2>/dev/null)" == zlib ]] && HAS_ZLIB=true
+
```

Comment:
Albeit this is certainly cool, it is problematic on systems whose OpenSSL don't have zlib support:

```
prompt> echo -e "\x78\x9C\xAB\xCA\xC9\x4C\xE2\x02\x00\x06\x20\x01\xBC" | openssl zlib -d 2 > /dev/null
openssl:Error: 'zlib' is an invalid command.

Standard commands
asn1parse         ca                ciphers           crl               
[..]
```

I suspect MacOS is one of them

Reply:
Hi Dirk,

The above code worked for me, even with versions of OpenSSL that did not have zlib support. I believe the problem is the extra spaces in the test that you performed. By writing "2 >" rather than "2>", the output of stderr is not redirected to /dev/null. Even if OpenSSL has zlib support, the test you ran would not work, as it would treat "2" as a parameter to openssl rather than as part of the redirect command.

Reply:
Sorry. I tried it under an old FreeBSD but I forgot the default shell is /bin/csh :-/

A little bit strange though - as I pasted your code into the command line the paste was "1:1" but threw the error. When I pasted the code my alleged proof I used the cmd line repetition which then showed the blank inserted.